### ISSUE TITLE
css output is broken when `:global` is specified in the css selector.

### ISSUE THREAD

**Issue by Thiry1** [web-infra-dev/rspack#3611]
### System Info

 System:
    OS: macOS 13.4
    CPU: (8) arm64 Apple M2
    Memory: 76.59 MB / 24.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 18.16.0 - ~/.nodebrew/current/bin/node
    Yarn: 1.22.19 - ~/.nodebrew/current/bin/yarn
    npm: 9.5.1 - ~/.nodebrew/current/bin/npm
  Browsers:
    Chrome: 114.0.5735.133
    Safari: 16.5
  npmPackages:
    @rspack/cli: ^0.2.3 => 0.2.3 

### Details

When build the following scss, the css output is broken.
input: 
```scss
.my-class:global {
  font-size: 100px;
}
```
output:
```css
 {
  font-size: 100px;
}
```
As you can see, the selector is missing. But js has the selector correctly.
https://github.com/Thiry1/rspack-bug-repro/blob/main/rspack-dist/main.js#L13

However, putting a space between the selector and `:global` seems to work.
input: 
```scss
.my-class :global {
  font-size: 100px;
}
```
output:
```css
.b2fabca1ec7065ff  {
  font-size: 100px;
}
```

I don't know if not putting a space between the selector and :global is a valid syntax, but webpack is able to output it with no problem.

### Reproduce link

https://github.com/Thiry1/rspack-bug-repro

### Reproduce Steps

1. run `npm run build`
2. see `rspack-dist` directory.

**Comment by hyf0** [comment]
@ahabhgk Do we support patterns like this?

```css
.my-class:global {
  font-size: 100px;
}
```

**Comment by ahabhgk** [comment]
Looks like swc-css doesn't support this

**Comment by hyf0** [comment]
@Thiry1 
We use `swc_css` to parse `css` and `css modules` in Rspack, so the error is actually thrown by `swc`.

For a workaround, you could use `css-loader` and `style-loader` to parse the input without errors.

If you manage to parse the input with `css-loader` and `style-loader`, I will create an issue in `swc` repo to support this pattern.

**Comment by hyf0** [comment]
> Looks like swc-css doesn't support this

@ahabhgk is this pattern valid `css modules`.

**Comment by ahabhgk** [comment]
I think so

**Comment by Thiry1** [comment]
@hyf0 Thank you for reply.

>For a workaround, you could use css-loader and style-loader to parse the input without errors.

I tried it and it seems to output the correct css without error.
rspack config: https://github.com/Thiry1/rspack-bug-repro/blob/83b6059b51ad87425206ace54eef7e4d31154b4f/rspack.config.js#L9-L20
output:
https://github.com/Thiry1/rspack-bug-repro/blob/83b6059b51ad87425206ace54eef7e4d31154b4f/rspack-dist/main.js#L19

**Comment by hyf0** [comment]
> @hyf0 Thank you for reply.
> 
> > For a workaround, you could use css-loader and style-loader to parse the input without errors.
> 
> I tried it and it seems to output the correct css without error. rspack config: https://github.com/Thiry1/rspack-bug-repro/blob/83b6059b51ad87425206ace54eef7e4d31154b4f/rspack.config.js#L9-L20 output: https://github.com/Thiry1/rspack-bug-repro/blob/83b6059b51ad87425206ace54eef7e4d31154b4f/rspack-dist/main.js#L19

Thank you. It should be an unsupported feature of `swc_css`. I would create an issue in swc repo.

**Comment by hyf0** [comment]
It seems fixed at https://github.com/swc-project/swc/pull/7450 by @suxin2017. Great work to @suxin2017.

**Comment by hyf0** [comment]
Need to wait #3583.

**Comment by Thiry1** [comment]
I have confirmed the bug has been fixed in the 0.2.5-canary-db1be92-20230706140010.
Thank you!
